src/asmcnc/apps/drywall_cutter_app/widget_drywall_shape_display.py

```diff
@@ -298,14 +411,209 @@ def rotation_required(self):
         else:
             return False
 
-    def poll_position(self, dt):
-        current_x = round(abs(self.m.mpos_x()), 2)
-        current_y = round(abs(self.m.mpos_y()), 2)
+    def remove_negatives(self, instance, value):
+        if value.startswith("-"):
+            # Stop user inputting negative values
+            instance.text = ""
+            value = ""
+        return value
+
+    def check_datum_and_extents(self, dt):
```

Comment:
This function is pretty heavy to look through, could this be broken out into separate functions?